File: Cpp/Add Two Numbers.cpp
#include <queue>

struct ListNode {
    int val;
    ListNode *next;
    ListNode(int x) : val(x), next(NULL) {}
};


ListNode *addTwoNumbers(ListNode *l1, ListNode *l2) {
    if(l1==NULL) return l2;
    if(l2==NULL) return l1;
    ListNode *node1=l1,*node2=l2;
    ListNode *prenode=NULL,*pre;
    int d=0;
    while(node1!=NULL&&node2!=NULL){
        if(prenode==NULL){
            ListNode *root=new ListNode(sizeof(ListNode));
            int cnt=d+node1->val+node2->val;
            d=cnt/10;
            root->val=cnt%10;
            prenode=root;
            pre=root;
        }
        else{
            int cnt=d+node1->val+node2->val;
            d=cnt/10;
            ListNode *root=new ListNode(sizeof(ListNode));
            root->val=cnt%10;
            prenode->next=root;
            prenode=prenode->next;
        }
        node1=node1->next;
        node2=node2->next;
    }
    if(node1==NULL){
        while(node2!=NULL){
            int cnt=d+node2->val;
            d=cnt/10;
            ListNode *root=new ListNode(sizeof(ListNode));
            root->val=cnt%10;
            prenode->next=root;
            prenode=prenode->next;
            node2=node2->next;
        }
    }
    else if(node2==NULL){
        while(node1!=NULL){
            int cnt=d+node1->val;
            d=cnt/10;
            ListNode *root=new ListNode(sizeof(ListNode));
            root->val=cnt%10;
            prenode->next=root;
            prenode=prenode->next;
            node1=node1->next;
        }
    }
    if(d>0){
        ListNode *root=new ListNode(sizeof(ListNode));
        root->val=d;
        prenode->next=root;
        prenode=prenode->next;
    }
    prenode->next=NULL;
    return pre;
}

File: Python/Add-Two-Numbers.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution:
    # @return a ListNode
    def addTwoNumbers(self, l1, l2):
        seq = []
        while l1 != None and l2 != None:
            seq.append(l1.val + l2.val)
            l1 = l1.next
            l2 = l2.next
        while l1 != None:
            seq.append(l1.val)
            l1 = l1.next
        while l2 != None:
            seq.append(l2.val)
            l2 = l2.next
        cnt = 0
        head = None
        node = None
        while len(seq) > 0:
            ans = seq.pop(0)
            if head == None:
                head = ListNode((ans + cnt) % 10)
                cnt = (ans + cnt) / 10
                node = head
            else:
                node1 = ListNode((ans + cnt) % 10)
                cnt = (ans + cnt) / 10
                node.next = node1
                node = node1
        if cnt > 0:
            node1 = ListNode(cnt)
            node.next = node1
            node = node1
        node.next = None
        return head